pkg/project/options.go

```diff
@@ -68,6 +69,10 @@ func FromConfig(c *Config) (*Project, error) {
 }
 
 func FunctionFromHandler(h string) (Function, error) {
+	if regexp.MustCompile(`[~^\[\]()#$*&!\s]`).MatchString(h) {
```

Comment:
How did you determine the list of characters to check for, are we sure this is the full list?

Reply:
The error is caused by tag names in docker - "A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and hyphens.".

Tested the ones included in the list caused the issue - it's possible there are more. I will try and find more info or spot-check chars like `@%+=|?<>, to see if they need to be added.

Reply:
does docker already have a regex for this that we can reuse?

Reply:
On my local, I was working on the docker image support. I created an image string parser util for docker (including the tag)
So you could re-use some of this to validate if you like, as it uses "github.com/docker/distribution/reference"

```
package utils

import (
	"bytes"

	"github.com/docker/distribution/reference"
)

func StringTrunc(s string, max int) string {
	if len(s) <= max {
		return s
	}

	return s[:max]
}

func JoinCamelCase(ss []string) string {
	res := ss[0]

	for i := 1; i < len(ss); i++ {
		word := ss[i]
		res += string(bytes.ToUpper([]byte{word[0]}))
		res += word[1:]
	}

	return res
}

type DockerImageMeta struct {
	Name string
	Tag string
}

type ParsedDocker interface {
	String() string
	Name() string
	Tag() string
}

func ParseDockerImage(image string) (DockerImageMeta, error) {
	meta := DockerImageMeta{}

	imageRef, err := reference.Parse(image)
    if err != nil {
        return meta, err
    }

	namedRef, ok := imageRef.(reference.Named)
    if !ok {
		return meta, err
    }

	meta.Name = reference.Path(namedRef)

	taggedRef, ok := imageRef.(reference.Tagged)
    if ok {
        meta.Tag = taggedRef.Tag()
    }

	return meta, nil
}
```